serialization.rst

```diff
@@ -43,24 +43,30 @@ interface, serialize as a **BSON document**. Keep only *public* properties,
 and ignore *protected* and *private* properties.
 
 If an object is of a class that implements the ``MongoDB\BSON\Serializable``
-interface, call ``bsonSerialize`` and use the returned associative array to
-store as properties of a **BSON document**. Not returning an array is an error
-and should throw an ``MongoDB\Driver\Exception\UnexpectedValueException``
-exception. It is valid to return a packed array, but it must also be stored as
-a **BSON document**.
+interface, call ``bsonSerialize`` and use the returned array or ``stdClass`` to
+serialize as a BSON document or array. The BSON type will be determined by the
+following:
+
+1. Root documents must be serialized as a BSON document.
+2. ``MongoDB\BSON\Persistable`` objects must be serialized as a BSON document.
+3. If ``bsonSerialize`` returns a packed array, serialize as a BSON array.
+4. If ``bsonSerialize`` returns a non-packed array or ``stdClass``, serialize as
+   a BSON document.
+5. If ``bsonSerialize`` did not return an array or ``stdClass``, throw an
+   ``MongoDB\Driver\Exception\UnexpectedValueException`` exception.
```

Comment:
The to-be-published-docs-from-this-document should include an example using (object) casting to force BSON document serialization.